### ISSUE TITLE
Use existing HTTP_PROXY and HTTPS_PROXY variables

### ISSUE THREAD

**Issue by bkimminich** [atom/atom#9907]
I am using Atom on a company Windows 7-laptop that is behind a proxy when at work and unproxied when at home. I am using e.g. ```git``` via ```MINGW64``` (the bash that comes with Git for Windows) and have a little script to toggle proxy settings on and off. It basically sets ```HTTP_PROXY``` and ```HTTPS_PROXY```, which is recognized by ```git```, ```bower``` and ```npm```.

Unfortunately ```apm``` doesn't seem to recongize those variables, as I get ```Request for package information failed: connect ETIMEDOUT``` when running ```apm update``` from ```MINGW64``` while behind corporate proxy, no matter if proxy settings are declared or not.

**Comment by bkimminich** [comment]
This is my ```.bashrc``` with the proxy toggle script:
```
function proxy_on(){
   export HTTP_PROXY="http://$PROXY_SERVER:$PROXY_PORT"
   export HTTPS_PROXY="$HTTP_PROXY" FTP_PROXY="$HTTP_PROXY" ALL_PROXY="$HTTP_PROXY" \
      NO_PROXY="localhost,127.0.0.1" \
      GIT_CURL_VERBOSE=1 GIT_SSL_NO_VERIFY=1
   echo -e "\nProxy-related environment variables set."
}
 
function proxy_off(){
   variables=( "HTTP_PROXY" "HTTPS_PROXY" "FTP_PROXY" "ALL_PROXY" \
      "NO_PROXY" "GIT_CURL_VERBOSE" "GIT_SSL_NO_VERIFY" )
   for i in "${variables[@]}"; do unset $i; done
   echo -e "\nProxy-related environment variables removed."
}

proxy_on
```

and this is my ```.atomrc```, but having this line or not doesn't make a difference:

```
strict-ssl = false
```

**Comment by thedaniel** [comment]
This was fixed in https://github.com/atom/apm/pull/447 and will be included in 1.3 - you can confirm with the beta release if you'd like - it's available at https://github.com/atom/atom/releases
